commit 69500c43b45f7155b72dcadad31cd55cda789c93
Author: Tom Tucker <tom@opengridcomputing.com>
Date:   Wed May 7 13:49:58 2008 -0500

    svcrdma: Set rqstp transport address in rdma_read_complete function
    
    The rdma_read_complete function needs to copy the rqstp transport address
    from the transport. Failure to do so can result in using the wrong
    authentication method for the RPC or bug checking if the rqstp address
    is not valid.
    
    Signed-off-by: Tom Tucker <tom@opengridcomputing.com>

diff --git a/net/sunrpc/xprtrdma/svc_rdma_recvfrom.c b/net/sunrpc/xprtrdma/svc_rdma_recvfrom.c
index 21a1e625ef03..c016f5ca0ce5 100644
--- a/net/sunrpc/xprtrdma/svc_rdma_recvfrom.c
+++ b/net/sunrpc/xprtrdma/svc_rdma_recvfrom.c
@@ -416,6 +416,7 @@ static int rdma_read_complete(struct svc_rqst *rqstp,
 
 	/* XXX: What should this be? */
 	rqstp->rq_prot = IPPROTO_MAX;
+	svc_xprt_copy_addrs(rqstp, rqstp->rq_xprt);
 
 	ret = rqstp->rq_arg.head[0].iov_len
 		+ rqstp->rq_arg.page_len